### ISSUE TITLE
aFreeRDP resource compilation issues on Ubuntu, Level 23 Android build tools

### ISSUE THREAD

**Issue by MrCsabaToth** [FreeRDP/FreeRDP#3031]
Details:
http://stackoverflow.com/questions/34555121/how-to-compile-afreerdp-on-linux

http://pastebin.com/Nc0mqGrQ

 [aapt] Generating resource IDs...
 [aapt] /home/csaba/GitRepos/FreeRDP/client/Android/appcompat_v7/res/values-v23/styles_base.xml:20: error: Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Button.Colored'.
 [aapt]
 [aapt] /home/csaba/GitRepos/FreeRDP/client/Android/appcompat_v7/res/values-v23/styles_base_text.xml:19: error: Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.Button.Inverse'.
 [aapt]

**Comment by MrCsabaToth** [comment]
Any workaround is welcome

**Comment by MrCsabaToth** [comment]
I backed up my current Android SDK and tried to go back in time. SDK download tgz 22.6.2 only provides target 21. But SDK download tgz 23.0.0 already wants to upgrade to 24. So I cannot get the system to a state, when I don't have the target 23 (which indirectly causes errors somehow - more correctly, the level 24's compatibility facility for target 23 (which satisfies target 21)), but have the target 21.
What to do as a workaround?

**Comment by MrCsabaToth** [comment]
I started with the 23.0 SDK tgz, upgraded one round (not upgrading up to Android 6.0). Then the troublesome xml files are copied over from the Android SDK. So I deleted that values-v23 folder from the SDK. I also modified the AndroidManifest.xml of the aFreeRDP and set minSdkVersion to 17 from 14. The directory deletion certainly helped, I'm not sure if the other two steps contributed, but I have an apk now. I consider this an ugly hack, only a workaround though.

**Comment by MrCsabaToth** [comment]
Update: I started from a totally updated Android SDK, and the minSdkVersion bump from 14 to 17 also doesn't matter. The only thing which made the error go away is the invasive removal of that values-v23 directory from inside the Android SDK.

**Comment by akallabeth** [comment]
@MrCsabaToth There have been significant improvements for the FreeRDP android build, is this still an issue for you?

**Comment by MrCsabaToth** [comment]
It's much better now! I updated my SO entry in two places some weeks ago following the new guides, now I added this to even two more places! Thanks you! Now I just have to integrate cardboard.jar into the project to complete continue with my project. Probably the culprit of this original issue (error: Error retrieving parent for item: No resource found that matches the given name...) is an independent problem, and it's rather Google's fault I believe.
